service/src/main/java/uk/gov/hmcts/reform/fpl/service/DocumentService.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisOrder;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DocumentService {
+    private final DocmosisDocumentGeneratorService documentGeneratorService;
+    private final UploadDocumentService uploadDocumentService;
+
+    public <T extends DocmosisOrder> Document getDocumentFromDocmosisOrderTemplate(T templateData,
+                                                                                   DocmosisTemplates template) {
+        DocmosisDocument document = documentGeneratorService.generateDocmosisDocument(templateData, template);
+
+        String documentTitle = getDocumentTitle(templateData.getDraftbackground(), document);
+
+        return uploadDocumentService.uploadPDF(document.getBytes(), documentTitle);
+    }
+
+    private String getDocumentTitle(String draftBackground, DocmosisDocument document) {
+        if (draftBackground == null) {
+            return document.getDocumentTitle();
+        }
+        return document.getDraftDocumentTile();
```

Comment:
could be simplified to 
```suggestion
  return draftBackground == null ? document.getDocumentTitle() : document.getDraftDocumentTile();
```
or maybe even pass the result of draftBackground == null as a parameter instead of the string